feature-libs/checkout/b2b/core/http-interceptors/bad-request/bad-cost-center-request.handler.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from '@angular/core';
+import { HttpErrorResponse, HttpRequest } from '@angular/common/http';
+import {
+  ErrorModel,
+  GlobalMessageType,
+  GlobalMessageService,
+  HttpErrorHandler,
+  HttpResponseStatus,
+  Priority,
+  TranslationService,
+} from '@spartacus/core';
+import { ResponseError } from './bad-cost-center-request.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class BadCostCenterRequestHandler extends HttpErrorHandler {
+  responseStatus = HttpResponseStatus.BAD_REQUEST;
+
+  constructor(
+    protected globalMessageService: GlobalMessageService,
+    protected translationService: TranslationService
+  ) {
+    super(globalMessageService);
+  }
+
+  getPriority(): Priority {
+    return Priority.NORMAL;
+  }
+
+  hasMatch(errorResponse: HttpErrorResponse): boolean {
+    return (
+      super.hasMatch(errorResponse) &&
+      this.getErrors(errorResponse).some(this.isEntityValidationError) &&
+      this.isCostCenterRequest(errorResponse)
+    );
+  }
+
+  handleError(_request: HttpRequest<any>, response: HttpErrorResponse): void {
+    if (this.getErrors(response).some((e) => this.isEntityValidationError(e))) {
+      this.translationService
+        .translate('checkoutB2B.invalidCostCenter')
+        .subscribe((result: string) => {
+          this.globalMessageService.add(
```

Comment:
You can pass keys to the `globalMessage.add` method like: 
`this.globalMessage.add({ key: 'checkoutB2B.invalidCostCenter' }, GlobalMessageType.MSG_TYPE_CONFIRMATION);`. You don't need to make a subscription to the translation service.